kahypar/io/hypergraph_io.h

```diff
@@ -187,17 +208,52 @@ static inline void readHypergraphFile(const std::string& filename,
   }
 }
 
+static inline void validateAndPrintErrors(const HypernodeID num_hypernodes, const HyperedgeID num_hyperedges,
+                                          const size_t* hyperedge_indices, const HypernodeID* hyperedges,
+                                          const HyperedgeWeight* hyperedge_weights, const HypernodeWeight* vertex_weights,
+                                          const std::vector<size_t> line_numbers,
+                                          std::vector<HyperedgeID>& ignored_hes,
+                                          std::vector<size_t>& ignored_pins,
+                                          const bool promote_warnings_to_errors) {
+    ErrorList errors;
+    HighResClockTimepoint start = std::chrono::high_resolution_clock::now();
+    validate::validateInput(num_hypernodes, num_hyperedges, hyperedge_indices, hyperedges,
+                            hyperedge_weights, vertex_weights, &errors, &ignored_hes, &ignored_pins);
+    validate::printErrors(num_hypernodes, num_hyperedges, errors, line_numbers, promote_warnings_to_errors);
+    if (validate::containsFatalError(errors, promote_warnings_to_errors)) {
+      exit(1);
+    }
+    HighResClockTimepoint end = std::chrono::high_resolution_clock::now();
+    Timer::instance().add(Timepoint::input_validation,
+                          std::chrono::duration<double>(end - start).count());
+}
 
 static inline Hypergraph createHypergraphFromFile(const std::string& filename,
-                                                  const PartitionID num_parts) {
+                                                  const PartitionID num_parts,
+                                                  bool validate_input = true,
+                                                  bool promote_warnings_to_errors = false) {
```

Comment:
Would it be possible to avoid `default` parameter values?

Reply:
Yes. The function is used by quite a lot of tests and I wanted to avoid updating all of them, but I can also just do it^^

Reply:
That would be very much appreciated. 🙂 Default values are always a source for errors.